src/main/java/org/terasology/launcher/TerasologyLauncher.java

```diff
@@ -101,22 +99,19 @@ public void start(final Stage initialStage) {
         showSplashStage(initialStage, launcherInitTask);
         Thread initThread = new Thread(launcherInitTask);
 
-        launcherInitTask.setOnSucceeded(new EventHandler<WorkerStateEvent>() {
-            @Override
-            public void handle(final WorkerStateEvent workerStateEvent) {
-                try {
-                    LauncherConfiguration config = launcherInitTask.getValue();
-                    if (config == null) {
-                        throw new LauncherStartFailedException("Launcher configuration was `null`.");
-                    } else if (config instanceof NullLauncherConfiguration) {
-                        logger.info("Closing the launcher ... (empty configuration, probably due to update)");
-                        Platform.exit();
-                    } else {
-                        showMainStage(config);
-                    }
-                } catch (IOException | LauncherStartFailedException e) {
-                    openCrashReporterAndExit(e);
+        launcherInitTask.setOnSucceeded(workerStateEvent -> {
+            try {
+                LauncherConfiguration config = launcherInitTask.getValue();
+                if (config == null) {
+                    throw new LauncherStartFailedException("Launcher configuration was `null`.");
+                } else if (config instanceof NullLauncherConfiguration) {
+                    logger.info("Closing the launcher ... (empty configuration, probably due to update)");
+                    Platform.exit();
+                } else {
+                    showMainStage(config);
                 }
+            } catch (IOException | LauncherStartFailedException e) {
+                openCrashReporterAndExit(e);
```

Comment:
Are we sure this does the same?
Previous implementation of `EventHandler` seems to me as if the `workerStateEvent` may not yet be there when the `launcherInitTask` succeeds? But maybe I'm reading this wrong...

Reply:
This just uses a different syntax (lambda expression) for creating the handler for the `onSucceeded` event. As `handle(WorkerStateEvent event)` is the only method that needs to be implemented we can use the simplified syntax here.

```java
new EventHandler<WorkerStateEvent>() {
	@Override
    public void handle(final WorkerStateEvent workerStateEvent) {
        // do something
    }
}
```

Is replaced by 
```java
workerStateEvent -> {
    // do something
}
```

Reply:
Alright, so the new one also kind of "waits" for a `workerStateEvent` to come in?